index.js

```diff
@@ -1979,19 +2002,21 @@ module.exports = class Autobase extends ReadyResource {
   }
 
   async _flushIndexes () {
+    if (this._indexedLength === this.system.core.flushedLength) return true
+
     let complete = true
     this._updatingCores = false
 
-    for (const core of this._viewStore.opened.values()) {
-      if (!await core.flush()) complete = false
-    }
+    const { views } = await this.system.getIndexedInfo(this._indexedLength)
 
-    // updates emitted sync
     for (const core of this._viewStore.opened.values()) {
-      if (core.indexing === 0) continue
-      const indexing = core.indexing
-      core.indexing = 0
-      core._onindex(indexing)
+      const index = core.systemIndex
+      if (!core._isSystem() && (index === -1 || index >= views.length)) continue
+
+      const length = core._isSystem() ? this._indexedLength : views[index].length
+
+      complete &&= await core.flush(length)
```

Comment:
you mean this and not `if (!(await core.flush())) complete = false` like before?

Reply:
yeh do you prefer it written that way?

Reply:
ahh ofc it won't execute